### PR TITLE
feat: create signup component (step 2)

### PR DESCRIPTION
**Pull request by nadiia-tsytsylina**
## Describe your changes

Create a SignUpSecond component.
Made form with validation on inputs ( only required and limit of characters yet). Add DatePicker to date of birth.

Снимок экрана 2023-11-09 192300
Снимок экрана 2023-11-09 192426
Снимок экрана 2023-11-09 192443

## Issue ticket code (and/or) and link

- [Link to JIRA ticket](https://homecareaid.atlassian.net/browse/AC-5)

### **General**

- [x] Assigned myself to the PR
- [x] Assigned the appropriate labels to the PR
- [x] Assigned the appropriate reviewers to the PR
- [ ] Updated the documentation
- [x] Performed a self-review of my code
- [x] Types for input and output parameters
- [x] Don't have "any" on my code
- [ ] Used the try/catch pattern for error handling
- [x] Don't have magic numbers
- [x] Compare only with constants not with strings
- [x] No ternary operator inside the ternary operator
- [x] Don't have commented code
- [x] no links in the code, env links should be in env file (for example: server url), constant links (for example default avatar URL) should be in constant file.
- [x] Used camelCase for variables and functions
- [x] Date and time formats are on the constants
- [x] Functions are public only if it's used outside the class
- [x] No hardcoded values
- [ ] covered by tests
- [x] Check your commit messages meet the [conventional commit format](https://www.conventionalcommits.org/en/v1.0.0/).

### Frontend

- [x] Components and business logic are separated
- [x] Colors, Font Size, and Font Name is on the theme or in the constants
- [x] No text in the components, use i18n approach
- [x] No inline styles
- [x] Imports are absolute
- [x] Attach a screenshot if PR has visual changes.

### CODE REVIEW

**Comment by lyudad** [comment]
Screenshot 2023-11-09 at 20 45 16

**Comment by nadiia-tsytsylina** [comment]
## Describe your changes

Update styles. 
Had some problems with phone number input. 
I tried to use [react-phone-number-input](https://www.npmjs.com/package/react-phone-number-input) . But appeared many problems with styles and using it with react-hook-form. After I found [mui-tel-input](https://viclafouch.github.io/mui-tel-input/docs/getting-started/) . But after installing an error occured "Cannot use import statement outside a module". Although I did imports as in documentation. Maybe you can give me some advice, what library better to use?
So for now I temporary did phone validation only for USA (number shoud begin from +1 and contain 11 numbers)

Снимок экрана 2023-11-10 151556
Снимок экрана 2023-11-10 151636
Снимок экрана 2023-11-10 151646
Снимок экрана 2023-11-10 151655

## Issue ticket code (and/or) and link

- [Link to JIRA ticket](https://homecareaid.atlassian.net/browse/AC-5)
update styles. Alse add validation in all fields.

**Comment by nadiia-tsytsylina** [comment]
last styles updates. Colors will be as in Figma after you approve my theme commit
Снимок экрана 2023-11-10 190638
Снимок экрана 2023-11-10 190721

**Comment by nadiia-tsytsylina** [comment]
last styles updates. Colors will be as in Figma after you approve my theme PR
Added phone input with countries selection. Used [mui-tel-input](https://viclafouch.github.io/mui-tel-input/docs/getting-started/). But I still have error  "Cannot use import statement outside a module", if I refresh the page. Also this library has own validation for different countries. For example, it's forbidden to begin from 0 for Ukraine, but for the USA and Canada it's forbidden to begin from 1, not 0. 

Снимок экрана 2023-11-11 140642
Снимок экрана 2023-11-11 140749
Снимок экрана 2023-11-11 140842
Снимок экрана 2023-11-11 140850

**Comment by lyudad** [comment]
> last styles updates. Colors will be as in Figma after you approve my theme PR Added phone input with countries selection. Used [mui-tel-input](https://viclafouch.github.io/mui-tel-input/docs/getting-started/). But I still have error "Cannot use import statement outside a module", if I refresh the page
> 
> Снимок экрана 2023-11-11 170628 Снимок экрана 2023-11-11 170837 Снимок экрана 2023-11-11 170651 Снимок экрана 2023-11-11 170749 Снимок экрана 2023-11-11 170807

I have not faced issue with libraries you mentioned, you can create own validator as we support now only 2 countries. I thing there are a lot of regular expression in the internet that could help you validate numer

**Comment by nadiia-tsytsylina** [comment]
fix: phone validation and styles.
changed styles of inputs. forbided users to begin phone number from 0.

Снимок экрана 2023-11-14 123132
Снимок экрана 2023-11-14 123154

File: src/components/sign-up-second/SignUpSecond.tsx [hunk 1]
```diff
@@ -0,0 +1,171 @@
+// @mui
```
Comment (lyudad) [inline src/components/sign-up-second/SignUpSecond.tsx:1]: why this comment needed?

File: src/components/sign-up-second/validation.ts [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+  const firstNameInvalid: string = translate('signUpSecondForm.firstNameInvalid');
+  const lastNameInvalid: string = translate('signUpSecondForm.lastNameInvalid');
+  const emailLengthInvalid: string = translate('signUpSecondForm.emailLengthInvalid');
+  const emailInvalid: string = translate('signUpSecondForm.emailInvalid');
+  const phoneLengthInvalid: string = translate('signUpSecondForm.phoneLengthInvalid');
+  const firstNameRequired: string = translate('signUpSecondForm.firstNameRequired');
+  const lastNameRequired: string = translate('signUpSecondForm.lastNameRequired');
+  const emailRequired: string = translate('signUpSecondForm.emailRequired');
+  const phoneRequired: string = translate('signUpSecondForm.phoneRequired');
+  const birthDateRequired: string = translate('signUpSecondForm.birthDateRequired');
```
Comment (lyudad) [inline src/components/sign-up-second/validation.ts:36]: why we need separate content variables for translation?

**Comment by nadiia-tsytsylina** [comment]
fix: deleted unnecessary comments and variables

**Comment by nadiia-tsytsylina** [comment]
add redux slice for personal details

File: src/components/sign-up-second/SignUpSecond.tsx [hunk 3]
```diff
@@ -0,0 +1,175 @@
...
+    setError,
+    clearErrors,
+    formState: { errors, isValid },
+  } = useForm<FormValues>({
+    resolver: yupResolver(signUpSecondSchema),
+    mode: 'onBlur',
+  });
+
+  const onSubmit = handleSubmit(({ firstName, lastName, email, phone, birthDate, isOpen }) => {
+    const dateToString = birthDate.toLocaleDateString().padStart(10, '0');
```
Comment (lyudad) [inline src/components/sign-up-second/SignUpSecond.tsx:52]: https://www.notion.so/Git-rules-849f8ca8b3db4ded926d762c6fff8798 1.1.4 лично мне не понятно почему 10, почему не 6 или 52

Comment (nadiia-tsytsylina) [reply src/components/sign-up-second/SignUpSecond.tsx:52]: исправила

**Pull request merged**